### ISSUE TITLE
Adapter List; Support for multiple adapters of the same type

### ISSUE THREAD

**Issue by robin-homberg** [Flagbit/Magento2-Flysystem#33]
For a lot of features that could be developed for the Magento2 Flysystem Module; And for the usability and customisability it would make a lot of sense to integrate a Flyssytem Adapter list into the Adminpanel of Magento2.
Features the List should be able to:
- Create New Adapters
- Edit Adapters
- Delete Adapters
- Test Adapter Connection
( - Show files linked to the current Adapter )

The Adapters should be stored in the database. There you should take care of encrypting the passwords for the api connections, the same way, like the magento store configuration does.
The storeconfiguration needs to be minidied afterwards. That you are able to select an adapter from the adapter list, which should be used by default inside media modals, instead of providing the adapter type, which can be configured inside the store configuration, like it is right now.

To migrate 0.2.x Flysystem versions to the new configuration of 0.3.x versions, it would be very nice to have a setup script which transforms the old adapters configurations into the new ones.

This feature would be also great to have, before starting with other new features like #24 .

**Comment by robin-homberg** [comment]
In Work on branch FLYS-20.
This changes wont be compatible with custom changes made for Magento2-Flysystem <=0.2.x, while it will increase the customizability of the module. Thats why this feature will release with the version 0.3.0..
The official Magento2-Flysystem Repositories from Flagbit like Magento2-Flysystem-S3 will be updated according to the changes after the release of version 0.3.0.
